src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/SynthesizedClosureMethod.cs

```diff
@@ -99,6 +100,11 @@ internal sealed class SynthesizedClosureMethod : SynthesizedMethodBaseSymbol, IS
             }
 
             AssignTypeMapAndTypeParameters(typeMap, typeParameters);
+
+            if (ReturnType.ContainsNullableReferenceTypes())
+            {
+                this.DeclaringCompilation.EnsureNullableAttributeExists(diagnostics, Locations[0], modifyCompilation: false);
```

Comment:
>this.DeclaringCompilation.EnsureNullableAttributeExists(diagnostics, Locations[0], modifyCompilation: false); [](start = 16, length = 109)

It is not clear why do we have to do this here. Also, why only for return types, what about parameters? Why we don't do this for ref readonly here?

Reply:
Moved to `LocalFunctionSymbol`.